### PR TITLE
fix(heatmap): adjust pageSize based available chart height

### PR DESCRIPTION
**Pull request by markov00**
## Summary

This fix the issue with wrongly computed pageSize: when computing the pageSize parameter we should consider the x axis vertical space as space not available for the rendering rows. 

To test:
- copy the PR playground file into a master branch, reduce the screen width and the top most chart will disappear
- on the PR instead, the chart will renderer correctly

### CODE REVIEW

File: src/chart_types/heatmap/state/selectors/get_grid_full_height.ts [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    const xAxisHeight = visible ? fontSize : 0;
+    const totalVerticalPadding = padding * 2;
+    let legendHeight = 0;
+    if (showLegend && (legendPosition === Position.Top || legendPosition === Position.Bottom)) {
+      legendHeight = maxLegendHeight ?? legendSize.height;
+    }
+    const verticalRemainingSpace = containerHeight - xAxisHeight - totalVerticalPadding - legendHeight;
+
+    // compute the grid cell height
+    const gridCellHeight = getGridCellHeight(yValues, grid, verticalRemainingSpace);
```
Comment (darnautov) [inline src/chart_types/heatmap/state/selectors/get_grid_full_height.ts:63]: nit: as `getGridCellHeight` is only used here maybe it's worth moving from the `viewmodel`

**Review by darnautov — APPROVED** [review]
Tested and LGTM

File: src/chart_types/heatmap/state/selectors/get_grid_full_height.ts (hunk 1 above)
Comment (markov00) [reply src/chart_types/heatmap/state/selectors/get_grid_full_height.ts:63]: done https://github.com/elastic/elastic-charts/pull/849/commits/b1d675015243501b02e1818f6cebf17e79a8fd0f

File: src/chart_types/heatmap/state/selectors/get_grid_full_height.ts [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+    { xAxisLabel: { padding, visible, fontSize }, grid, maxLegendHeight },
+    { yValues },
+  ): GridHeightParams => {
+    const xAxisHeight = visible ? fontSize : 0;
+    const totalVerticalPadding = padding * 2;
+    let legendHeight = 0;
+    if (showLegend && (legendPosition === Position.Top || legendPosition === Position.Bottom)) {
+      legendHeight = maxLegendHeight ?? legendSize.height;
+    }
+    const verticalRemainingSpace = containerHeight - xAxisHeight - totalVerticalPadding - legendHeight;
```
Comment (darnautov) [inline src/chart_types/heatmap/state/selectors/get_grid_full_height.ts:60]: we should probably check for `xAxisLabel.visible`

**Pull request merged**